var/spack/repos/builtin/packages/sim-model/package.py

```diff
@@ -139,23 +156,34 @@ def _install_src(self, prefix):
         """Copy original and translated c mods
         """
         arch = os.path.basename(self.neuron_archdir)
+        mkdirp(prefix.lib.mod, prefix.lib.hoc, prefix.lib.python)
         copy_all('mod', prefix.lib.mod)
         copy_all('hoc', prefix.lib.hoc)
+        if os.path.isdir('python'):  # Recent neurodamus
+            copy_all('python', prefix.lib.python)
 
-        for cmod in find(arch, "*.c", recursive=False):
+        for cmod in find(arch, '*.c', recursive=False):
             shutil.move(cmod, prefix.share.modc)
 
-        if self.spec.satisfies('+coreneuron'):
-            shutil.move(join_path(arch, 'core/mod2c'), prefix.share.modc_core)
+    def _setup_environment_common(self, spack_env, run_env):
+        spack_env.unset('LC_ALL')
+        # Remove LD_LIB_PATHs
+        to_rem = ('LD_LIBRARY_PATH', 'DYLD_LIBRARY_PATH', 'DYLD_FALLBACK_LIBRARY_PATH')
+        run_env.env_modifications = [envmod for envmod in run_env.env_modifications
+                                     if envmod.name not in to_rem]
+        if os.path.isdir(self.prefix.lib.hoc):
+            run_env.prepend_path('HOC_LIBRARY_PATH', self.prefix.lib.hoc)
+        if os.path.isdir(self.prefix.lib.python):
+            run_env.prepend_path('PYTHONPATH', self.prefix.lib.python)
 
     def setup_environment(self, spack_env, run_env):
-        spack_env.unset('LC_ALL')
-        run_env.set('NRNMECH_LIB_PATH', self.spec.prefix.lib.join('libnrnmech.so'))
-        run_env.set('BGLIBPY_MOD_LIBRARY_PATH', self.spec.prefix.lib.join('libnrnmech.so'))
+        self._setup_environment_common(spack_env, run_env)
+        for libnrnmech_name in find(self.prefix.lib, 'libnrnmech*.so', recursive=False):
+            run_env.prepend_path('BGLIBPY_MOD_LIBRARY_PATH', libnrnmech_name)
```

Comment:
Can BGLIBPY_MOD_LIBRARY_PATH have multiple paths?

Reply:
yep, by design we should support it. neurodamus-py already supports multiple libs